mirgecom/euler.py

```diff
@@ -144,35 +143,28 @@ def euler_operator(discr, eos, boundaries, q, t=0.0):
         Agglomerated object array of DOF arrays representing the RHS of the Euler
         flow equations.
     """
-    vol_flux = inviscid_flux(discr, eos, q)
-    dflux = discr.weak_div(vol_flux)
-
-    interior_face_flux = _facial_flux(
-        discr, eos=eos, q_tpair=interior_trace_pair(discr, q))
-
-    # Domain boundaries
-    domain_boundary_flux = sum(
-        _facial_flux(
-            discr,
-            q_tpair=boundaries[btag].boundary_pair(discr,
-                                                   eos=eos,
-                                                   btag=btag,
-                                                   t=t,
-                                                   q=q),
-            eos=eos
-        )
-        for btag in boundaries
-    )
-
-    # Flux across partition boundaries
-    partition_boundary_flux = sum(
-        _facial_flux(discr, eos=eos, q_tpair=part_pair)
-        for part_pair in cross_rank_trace_pairs(discr, q)
-    )
-
-    return discr.inverse_mass(
-        dflux - discr.face_mass(interior_face_flux + domain_boundary_flux
-                                + partition_boundary_flux)
+    vol_weak = discr.weak_div(inviscid_flux(discr=discr, eos=eos, cv=cv).join())
+
+    boundary_flux = (
+        _facial_flux(discr=discr, eos=eos, cv_tpair=interior_trace_pair(discr, cv))
+        + sum(
+            _facial_flux(discr, eos=eos, cv_tpair=part_pair)
+            for part_pair in cross_rank_trace_pairs(discr, cv))
+        + sum(
+            _facial_flux(
+                discr=discr,
+                cv_tpair=boundaries[btag].boundary_pair(discr,
+                                                        eos=eos,
+                                                        btag=btag,
+                                                        t=t,
+                                                        cv=cv),
+                eos=eos
+            )
+            for btag in boundaries)
+    ).join()
+
+    return split_conserved(
```

Comment:
Is this `join`/`split` pair here because `inverse_mass` and `face_mass` don't understand array containers? Let's teach them! I feel like we should ultimately be able to fire all uses of `{split,join}_conserved` from the code base.

Reply:
(cc @thomasgibson )

Reply:
Sounds like a good plan. Let's get https://github.com/inducer/grudge/pull/74 in ASAP so we can build on top of it. I'm keeping it regularly updated with all the new changing parts.

Reply:
Yes, and this one just above:

```python
    vol_weak = discr.weak_div(inviscid_flux(discr=discr, eos=eos, cv=cv).join())
```